### PR TITLE
Select by the source of object

### PR DESCRIPTION
**Pull request by lukaszachy**
E.g. Select all objects which use data defined in a particular file

### CODE REVIEW

File: fmf/base.py [hunk 1]
```diff
@@ -558,6 +562,9 @@ def prune(self, whole=False, keys=None, names=None, filters=None,
             if names and not any(
                     [re.search(name, node.name) for name in names]):
                 continue
+            # Select nodes defined by any of the source files
+            if sources and not bool(sources.intersection(node.sources)):
```
Comment (psss) [inline fmf/base.py:566]:
```suggestion
            if sources and not sources.intersection(node.sources):
```

Or is there a specific reason why explicitly convert to `bool` here?

**Review by psss — APPROVED** [review]
Thanks for writing this! Looks very good. Added some trivial adjustments and renamed duplicate test method (going to enable `flake8` to catch similar typos sooner).

File: fmf/base.py (hunk 1 above)
Comment (lukaszachy) [reply fmf/base.py:566]: Frankly I have not idea.

Comment (psss) [reply fmf/base.py]: Ok, removed then.

**Pull request merged**